### PR TITLE
No ability to exit out of MatchingFundsModal

### LINKED ISSUE #249: No ability to exit out of MatchingFundsModal
Clicked to "Add funds" to the matching pool:

Image 2021-08-10 at 12 55 02 PM

Modal should be closable, same UI as WalletModal component.

### PR DESCRIPTION
**Pull request by corwintines**
Fixes: https://github.com/ethereum/clrfund/issues/249

What changed:
- If the user doesnt have a wallet connected, it will just prompt the WalletModal for a connection now instead of prompting for them to click connect. This keeps functionality the same across different parts of the app when we get users to connect their wallet.

### CODE REVIEW

**Review by pettinarip — APPROVED** [review]
Great!

File: vue-app/src/views/ProjectList.vue [hunk 1]
```diff
@@ -216,7 +218,14 @@ export default class ProjectList extends Vue {
 
   addMatchingFunds(): void {
     if (!this.$store.state.currentUser) {
-      return
+      return this.$modal.show(
```
Comment (samajammin) [inline vue-app/src/views/ProjectList.vue]: I think this entire method can be removed? I don't believe `addMatchingFunds` is used here.

**Review by samajammin — CHANGES_REQUESTED** [review]
Looks mostly good!

One bug I noticed is on closing the modal, it keeps popping back up:

https://user-images.githubusercontent.com/8097623/129967395-1e15154b-dd89-4366-b1e5-dcd69ed962b9.mp4

I think we want to change the `closed` callback to not trigger the function again?

**Comment by samajammin** [comment]
A couple comments to address. Also please fix conflicts 🙏 - thanks!

**Comment by corwintines** [comment]
@samajammin good catch, thank you. I added a check on the closed callback to differentiate between if it was closing the modal vs going to the next step where the MatchingFundsModal should be open.

**Review by samajammin — CHANGES_REQUESTED** [review]
Thanks for the quick changes!

One more small issue I noticed:

https://user-images.githubusercontent.com/8097623/129981848-03cb67a4-935e-4393-810b-51874b708eb0.mp4

When we display the warning (i.e. user has insufficient funds) we should also disable the button to avoid this scary errors like this.

**Review by wackerow — APPROVED** [review]
nice, lookin good

**Review by samajammin — APPROVED** [review]

**Pull request merged**